src/search/merge_and_shrink/merge_selector.h

```diff
@@ -1,22 +1,31 @@
 #ifndef MERGE_AND_SHRINK_MERGE_SELECTOR_H
 #define MERGE_AND_SHRINK_MERGE_SELECTOR_H
 
+#include "../utils/logging.h"
+
 #include <string>
 #include <vector>
 
 class TaskProxy;
 
+namespace options {
+class Options;
+}
+
 namespace merge_and_shrink {
 class FactoredTransitionSystem;
+
 class MergeSelector {
 protected:
+    mutable utils::LogProxy log;
+
     virtual std::string name() const = 0;
-    virtual void dump_specific_options() const {}
+    virtual void dump_specific_options() const = 0;
     std::vector<std::pair<int, int>> compute_merge_candidates(
         const FactoredTransitionSystem &fts,
         const std::vector<int> &indices_subset) const;
 public:
-    MergeSelector() = default;
+    explicit MergeSelector(const options::Options &options);
```

Comment:
here you add new functionality (configuring the mergeselector), don't you?
If yes, I accept, just want to understand that nothing strange happens here.

Reply:
Yes, see previous reply.